rotkehlchen/db/dbhandler.py

```diff
@@ -3479,3 +3490,91 @@ def get_associated_locations(self) -> Set[Location]:
         if cursor.fetchone()[0] >= 1:
             locations.add(Location.BALANCER)
         return locations
+
+    def save_history_events(self, history: List[HistoryEvent]) -> bool:
+        query_str = 'INSERT INTO history_events(identifier, event_identifier, sequence_index, timestamp, location, location_label, asset, amount, usd_value, notes, type, subtype) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);'  # noqa: E501
+        cursor = self.conn.cursor()
+        try:
+            cursor.executemany(query_str, [event.serialize_for_db() for event in history])
+        except DeserializationError as e:
+            log.error(f'Failed to insert kraken statking events in database. {str(e)}')
+            return False
+        self.update_last_write()
+        return True
+
+    def delete_history_events(
+        self,
+        location: Optional[Location] = None,
+        location_label: Optional[str] = None,
+    ) -> int:
```

Comment:
It's called in the tests to purge the information and test different responses from the ledger. Since in the future it will be exposed to the frontend i'll follow that approach